packages/flutter_goldens/lib/flutter_goldens.dart

```diff
@@ -69,222 +74,301 @@ abstract class FlutterGoldenFileComparator extends GoldenFileComparator {
-  /// Creates a [FlutterSkippingGoldenFileComparator] that will skip tests that
-  /// are not in the right environment for golden file testing.
-  FlutterSkippingGoldenFileComparator(Uri basedir) : super(basedir);
-
-  /// Creates a new [FlutterSkippingGoldenFileComparator] that mirrors the relative
-  /// path resolution of the default [goldenFileComparator].
-  static FlutterSkippingGoldenFileComparator fromDefaultComparator({
+///  * [GoldenFileComparator], the abstract class that
+///    [FlutterGoldenFileComparator] implements.
+///  * [FlutterSkiaGoldFileComparator], another
+///    [FlutterGoldenFileComparator] that uploads tests to the Skia Gold
+///    dashboard.
+///  * [FlutterPresubmitFileComparator], another
+///    [FlutterGoldenFileComparator] that tests golden images before changes are
+///    merged into the master branch.
+class FlutterLocalFileComparator extends FlutterGoldenFileComparator with LocalComparisonOutput {
+  /// Creates a [FlutterLocalFileComparator] that will test golden file
+  /// images against baselines requested from Flutter Gold.
+  ///
+  /// The [fs] and [platform] parameters useful in tests, where the default file
+  /// system and platform can be replaced by mock instances.
+  FlutterLocalFileComparator(
+    final Uri basedir,
+    final SkiaGoldClient skiaClient, {
+    final FileSystem fs = const LocalFileSystem(),
+    final Platform platform = const LocalPlatform(),
+  }) : super(
+    basedir,
+    skiaClient,
+    fs: fs,
+    platform: platform,
+  );
+
+  /// Creates a new [FlutterLocalFileComparator] that mirrors the
+  /// relative path resolution of the default [goldenFileComparator].
+  ///
+  /// The [goldens] and [defaultComparator] parameters are visible for testing
+  /// purposes only.
+  static Future<FlutterGoldenFileComparator> fromDefaultComparator(
+    final Platform platform, {
+    SkiaGoldClient goldens,
     LocalFileComparator defaultComparator,
-  }) {
+  }) async {
+
     defaultComparator ??= goldenFileComparator;
-    return FlutterSkippingGoldenFileComparator(defaultComparator.basedir);
+    final Directory baseDirectory = FlutterGoldenFileComparator.getBaseDirectory(
+      defaultComparator,
+      platform,
+    );
+    if (!baseDirectory.existsSync())
+      baseDirectory.createSync(recursive: true);
+
+    goldens ??= SkiaGoldClient(baseDirectory);
+    return FlutterLocalFileComparator(baseDirectory.uri, goldens);
   }
 
   @override
   Future<bool> compare(Uint8List imageBytes, Uri golden) async {
-    print('Skipping "$golden" test : Skia Gold is not available in this testing '
-      'environment and flutter/goldens repository comparison is only available '
-      'on Linux machines.'
+    golden = addPrefix(golden);
+    final List<int> goldenBytes = await skiaClient.getMasterBytes(golden.path);
+    if (goldenBytes.isEmpty) {
+      // There is no baseline for this test
+      print('No digests provided by Skia Gold for test: $golden. '
```

Comment:
If I add a new test can I see the image it creates somewhere locally to confirm that its what I expect?

Reply:
I'll add automatic output for this case.